client/state/slices/teamSlice.ts

```diff
@@ -11,71 +11,99 @@ import {
+  isPlayerEligibleForPosition,
+} from "@/utils/teamUtils";
 
 type SwapPayload = {
   fromPosition: SlotPosition;
   toPosition: SlotPosition;
 };
 
-// TODO: Take a look at refactoring some of these to not mutate state in the function
-// Possibly move them into teamUtils.ts
-const swapPlayers = <T extends LineupSlot | BenchSlot>(
-  arrayToSwap: T[],
-  fromSlot?: T,
-  toSlot?: T,
-) =>
-  arrayToSwap.map((o) => {
-    if (o.position === fromSlot?.position) {
-      return { ...o, player: toSlot?.player };
-    }
-    if (o.position === toSlot?.position) {
-      return { ...o, player: fromSlot?.player };
-    }
-    return o;
-  });
+// TODO: Possibly move them into teamUtils.ts
+const getNextBenchPosition = (bench: BenchSlot[]): BenchPosition =>
+  `BEN${bench.length + 1}` as BenchPosition;
+
+const addBenchSlot = (bench: BenchSlot[], player: Player) => {
+  const position = getNextBenchPosition(bench);
+  return [...bench, { position, player }];
+};
+
+const removeBenchSlot = (
+  bench: BenchSlot[],
+  position: BenchPosition,
+): BenchSlot[] => {
+  const filtered = bench.filter((slot) => slot.position !== position);
+  return filtered.map((slot, i) => ({
+    ...slot,
+    position: `BEN${i + 1}` as BenchPosition,
+  }));
+};
 
 const getSlotfromPosition = (
   bench: BenchSlot[],
   lineup: LineupSlot[],
   position: SlotPosition,
-) => {
-  if (position.toString().startsWith("BEN")) {
-    const benchSlot = findBenchSlotByPosition(bench, position as BenchPosition);
+): { lineupSlot: LineupSlot | null; benchSlot: BenchSlot | null } => {
+  if (isBenchPosition(position)) {
+    const benchSlot = findSlotFromPosition(bench, position);
     return {
       lineupSlot: null,
       benchSlot,
     };
   }
 
-  const lineupSlot = lineup.find((slot) => slot.position === position) ?? null;
+  const lineupSlot = findSlotFromPosition(lineup, position) ?? null;
   return {
     lineupSlot,
     benchSlot: null,
   };
 };
 
-const getNextBenchPosition = (bench: BenchSlot[]): BenchPosition =>
-  `BEN${bench.length + 1}` as BenchPosition;
+const swapPlayers = <T extends LineupSlot | BenchSlot>(
+  arrayToSwap: T[],
+  fromSlot?: T,
+  toSlot?: T,
+) =>
+  arrayToSwap.map((o) => {
+    if (o.position === fromSlot?.position) {
+      return { ...o, player: toSlot?.player };
+    }
+    if (o.position === toSlot?.position) {
+      return { ...o, player: fromSlot?.player };
+    }
+    return o;
+  });
 
-const findBenchSlotByPosition = (
-  bench: BenchSlot[],
-  position: BenchPosition,
-): BenchSlot | null => bench.find((slot) => slot.position === position) ?? null;
-
-const removeBenchSlot = (bench: BenchSlot[], position: BenchPosition) => {
-  const index = bench.findIndex((slot) => slot.position === position);
-  if (index !== -1) {
-    bench.splice(index, 1);
-    bench.forEach((slot, i) => {
-      slot.position = `BEN${i + 1}` as BenchPosition;
+const swapBenchAndLineup = (
+  lineupSlot: LineupSlot,
+  benchSlot: BenchSlot,
```

Comment:
do you think we should specify incoming similar to what you did when passing props

Reply:
Oh ok I misunderstood, thank you for explaining